src/main/java/org/hl7/davinci/priorauth/ClaimEndpoint.java

```diff
@@ -380,6 +412,84 @@ private boolean cancelClaim(String claimId, String patient) {
     return result;
   }
 
+  /**
+   * Helper function to complete the cascade aspect of a Claim cancel. When a
+   * Claim is cancelled this will cascade the cancel to all related Claims (both
+   * up and downstream). A cancel updates the database status as well as update
+   * the resource status
+   * 
+   * @param claimId - the id of the original Claim to be cancelled
+   */
+  private void cascadeCancel(String claimId) {
+    // Cascade delete shared maps
```

Comment:
// Cascade cancel shared maps ?

Reply:
The cascade has to go two ways (in the case the ID given has newer replacements and also replaces old ones). So these maps are used in both directions and are used as inputs to the database queries